source/common/router/router.cc

```diff
@@ -688,18 +688,25 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
 
 std::unique_ptr<GenericConnPool>
 Filter::createConnPool(Upstream::ThreadLocalCluster& thread_local_cluster) {
-  GenericConnPoolFactory* factory = nullptr;
+  static GenericConnPoolFactory* DefaultConnPoolFactory =
```

Comment:
This change adds a static which adds risk of hermetic issues in unit tests and other problems. It also creates it eagerly even if it won't be needed.

What benefits does this change have? Have you found the previous code was a bottleneck? If so, could we instead capture a pointer to this factory in another factory we can get access to from here?

I personally really like to avoid statics where possible.